src/main/java/net/liquidpineapple/pang/XmlHandler.java

```diff
@@ -0,0 +1,270 @@
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+
+import javax.imageio.ImageIO;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+/**
+ * Created by Jaap-Jan on 12-10-2016.
+ * Class which handles the parsing to and from XML files.
+ */
+public class XmlHandler {
+
+  /**
+   * Method which parses a Level into a xml file.
+   * @param level - Level to be parsed
+   * @param time - Time the level may take.
+   */
+  public static void createXmlFromLevel(LevelScreen level, int time) {
+    try {
+      DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+
+      // root elements
+      Document doc = docBuilder.newDocument();
+      Element rootElement = doc.createElement("level");
+      doc.appendChild(rootElement);
+
+      // background element
+      Element bg = doc.createElement("background");
+      bg.appendChild(doc.createTextNode("bg.png"));
+      rootElement.appendChild(bg);
+
+      //difficulty element
+      Element difficulty = doc.createElement("difficulty");
+      difficulty.appendChild(doc.createTextNode("1"));
+      rootElement.appendChild(difficulty);
+
+      // objects
+      Element objects = doc.createElement("objects");
+      rootElement.appendChild(objects);
+
+      // parse Balls
+      for (GameObject object : level.objectList) {
+        if (object instanceof Ball) {
+          Element ball = doc.createElement("ball");
+          objects.appendChild(ball);
+
+          Element position = doc.createElement("position");
+          ball.appendChild(position);
+
+          int xposInt = (int) (object.getXpos() + 0.5);
+          int yposInt = (int) (object.getYpos() + 0.5);
+
+          Element xelement = doc.createElement("x");
+          xelement.appendChild(doc.createTextNode(Integer.toString(xposInt)));
+          position.appendChild(xelement);
+          Element yelement = doc.createElement("y");
+          yelement.appendChild(doc.createTextNode(Integer.toString(yposInt)));
+          position.appendChild(yelement);
+
+          Element direction = doc.createElement("direction");
+          direction.appendChild(doc.createTextNode("LEFT_MOVEMENT"));
+          ball.appendChild(direction);
+
+          Element size = doc.createElement("size");
+          size.appendChild(doc.createTextNode(Integer.toString(((Ball) object).getBallSize())));
+          ball.appendChild(size);
+        }
+      }
+
+      //parse Players
+      for (GameObject object : level.objectList) {
```

Comment:
Same goes for this, I'd suggest methods like parsePlayer(Element objects) and parseBalls(Element objects)